src/psyclone/tests/gocean/transformations/globalstoargs_test.py

```diff
@@ -0,0 +1,260 @@
+                           api=API)
+    psy = PSyFactory(API).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    notkernel = invoke.schedule.children[0]
+    kernel = invoke.schedule.coded_kernels()[0]
+
+    # Monckeypatch resolve_deferred to avoid module searching and importing
+    # in this test. In this case we assume it is a REAL
+    for var in kernel.get_kernel_schedule().symbol_table.global_datasymbols:
+        def set_to_real(variable):
+            variable._datatype = DataType.REAL
+        monkeypatch.setattr(var, "resolve_deferred", lambda: set_to_real(var))
+
+    # Test with invalid node
+    with pytest.raises(TransformationError) as err:
+        trans.apply(notkernel)
+    assert ("The KernelGlobalsToArguments transformation can only be applied"
+            " to CodedKern nodes but found 'GOLoop' instead."
+            in str(err.value))
+
+    # Test transforming a single kernel
+    trans.apply(kernel)
+
+    # The transformation;
+    # 1) Has imported the symbol into the InvokeSchedule
+    assert invoke.schedule.symbol_table.lookup("rdt")
+    assert invoke.schedule.symbol_table.lookup("model_mod")
+    var = invoke.schedule.symbol_table.lookup("rdt")
+    container = invoke.schedule.symbol_table.lookup("model_mod")
+    assert var.is_global
+    assert var.interface.container_symbol == container
+
+    # 2) Has added the symbol as the last argument in the kernel call
+    assert isinstance(kernel.args[-1], Argument)
+    assert kernel.args[-1].name == "rdt"
+
+    # 3) Has converted the Kernel Schedule symbol into an argument which is
+    # in also the last position
+    ksymbol = kernel.get_kernel_schedule().symbol_table.lookup("rdt")
+    assert ksymbol.is_argument
+    assert kernel.get_kernel_schedule().symbol_table.argument_list[-1] == \
+        ksymbol
+    assert len(kernel.get_kernel_schedule().symbol_table.argument_list) == \
+        len(kernel.args) + 2  # GOcean kernels have 2 implicit arguments
+
+    # Check the kernel code is generated as expected
+    fwriter = FortranWriter()
+    kernel_code = fwriter(kernel.get_kernel_schedule())
+    assert "subroutine kernel_with_use_code(ji,jj,istep,ssha,tmask,rdt)" \
+        in kernel_code
+    assert "real, intent(inout) :: rdt" in kernel_code
+
+    # Check that the PSy-layer generated code now contains the use statement
+    # and argument call
+    generated_code = str(psy.gen)
+    assert "USE model_mod, ONLY: rdt" in generated_code
+    assert "CALL kernel_with_use_code(i, j, oldu_fld, cu_fld%data, " \
+           "cu_fld%grid%tmask, rdt)" in generated_code
+
+
+def test_globalstoargumentstrans_constant(monkeypatch):
+    ''' Check the KernelGlobalsToArguments transformation '''
+    from psyclone.transformations import KernelGlobalsToArguments
+    from psyclone.psyir.backend.fortran import FortranWriter
+
+    trans = KernelGlobalsToArguments()
+    assert trans.name == "KernelGlobalsToArguments"
```

Comment:
You already have these checks in the previous test :-)

Reply:
Removed from here